src/dotnet/APIView/APIViewWeb/Managers/ReviewManager.cs

```diff
@@ -475,9 +545,115 @@ private async Task<ReviewListItemModel> ToggleReviewApproval(ClaimsPrincipal use
             review.ChangeHistory = changeUpdate.ChangeHistory;
             review.IsApproved = changeUpdate.ChangeStatus;
             await _reviewsRepository.UpsertReviewAsync(review);
+            
+            // If this review was approved (first release), check if we should auto-approve namespace for related TypeSpec
+            if (changeUpdate.ChangeStatus && IsSDKLanguage(review.Language))
+            {
+                await CheckAndAutoApproveNamespaceForTypeSpec(review.PackageName, userId);
+            }
+            
             return review;
         }
 
+        /// <summary>
+        /// Check if all related SDK reviews are approved and if so, automatically approve namespace for TypeSpec review
+        /// </summary>
+        /// <param name="packageName"></param>
+        /// <param name="userId"></param>
+        /// <returns></returns>
+        private async Task CheckAndAutoApproveNamespaceForTypeSpec(string packageName, string userId)
+        {
+            // Get the base package name (e.g., "contoso" from "contoso.widget")
+            var packageBaseName = packageName?.Split('.').FirstOrDefault()?.ToLower();
+            if (string.IsNullOrEmpty(packageBaseName))
+                return;
+
+            try
+            {
+                // Find the TypeSpec review for this package
+                var typeSpecReview = await _reviewsRepository.GetReviewAsync("TypeSpec", packageName, false);
```

Comment:
This is not going to work when language package name is sent in request which won't match to the package name in DB for TypeSpec